tests/manage/mcg/test_mcg_resources_disruptions.py

```diff
@@ -132,19 +135,44 @@ def test_scale_down_rgw(self, scale_down_to):
         ],
     )
     def test_drain_mcg_pod_node(
-        self, node_drain_teardown, reduce_and_resume_cluster_load, pod_to_drain
+        self,
+        node_drain_teardown,
+        reduce_and_resume_cluster_load,
+        pod_to_drain,
+        timeout=30,
     ):
         """
-        Test drianage of nodes which contain NB resources
+        Test drainage of nodes which contain NB resources
 
         """
-        # Retrieve the relevant pod object
-        pod_obj = pod.Pod(
-            **pod.get_pods_having_label(
-                label=self.labels_map[pod_to_drain],
-                namespace=defaults.ROOK_CLUSTER_NAMESPACE,
-            )[0]
-        )
+        loop_cnt = 0
+        while True:
```

Comment:
We should use utils.py::TimeoutSampler or retry.py::retry instead for brevity, better maintainability and readability